keyboards/bastardkb/dilemma/4x6_4/4x6_4.c

```diff
@@ -57,3 +56,58 @@ bool encoder_update_kb(uint8_t index, bool clockwise) {
     return true;
 }
 #endif // ENCODER_ENABLE
+
+#ifdef RGB_MATRIX_ENABLE
+HSV _get_hsv_for_layer_index(uint8_t layer) {
+    switch (layer) {
+        case 1:
+            return (HSV){HSV_BLUE};
+        case 2:
+            return (HSV){HSV_AZURE};
+        case 3:
+            return (HSV){HSV_ORANGE};
+        case 4:
+            return (HSV){HSV_GREEN};
+        case 5:
+            return (HSV){HSV_TEAL};
+        case 6:
+            return (HSV){HSV_PURPLE};
+        case 7:
+        default:
+            return (HSV){HSV_RED};
+    };
+}
+
+// Layer state indicator
+bool rgb_matrix_indicators_advanced_kb(uint8_t led_min, uint8_t led_max) {
+    if (!rgb_matrix_indicators_advanced_user(led_min, led_max)) {
+        return false;
+    }
+    if (host_keyboard_led_state().caps_lock) {
```

Comment:
Add a comment to explain that we're changing the LED color of modifier keys to red when capslock is enable. Also worth mentioning that it's only working on Windows and Linux because host state is not passed down the keyboard on macOS (IIRC?).

Reply:
The docs mention no such restriction. A bit of googling reveals indicator state is per-keyboard on macos, so it won't work if another keyboard changes the lock state, but nothing implies it wouldn't work confined to one board.